lib/util/relative-module-resolver.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * Utility for resolving a module relative to another module
+ * @author Teddy Katz
+ */
+
+"use strict";
+
+const Module = require("module");
+const path = require("path");
+
+// Polyfill Node's `Module.createRequireFromPath` if not present (added in Node v10.12.0)
+const createRequireFromPath = Module.createRequireFromPath || (filename => {
+    const mod = new Module(filename, null);
+
+    mod.filename = filename;
+    mod.paths = Module._nodeModulePaths(path.dirname(filename)); // eslint-disable-line no-underscore-dangle
+    mod._compile("module.exports = require;", filename); // eslint-disable-line no-underscore-dangle
```

Comment:
Isn't it a bit heavy? I would have thought that `require.resolve(..., {paths: _nodeModulePaths(...)})` would have been enough?

Reply:
That would work for packages, but it would still resolve relative paths relative to `lib/util/relative-module-resolver.js` rather than relative to `filename`.

For example, if the folder `foo/` contains `bar.js` and `baz.js`, then

`require.resolve('./baz.js', {paths: Module._nodeModulePaths(path.resolve('./bar.js'))})` will fail to find `foo/baz.js`.

Reply:
Maybe it can be simplified with `path` module?

```js
function resolve(id, importerPath) {
    if (/^\.[./\\]/.test(id)) {
        id = path.resolve(importerPath, id)
    }
    return require.resolve(id, { paths: [importerPath] })
}
```

Reply:
Personally, I think the current code is a bit simpler than the version with the `path` module since we don't have to try to exactly replicate how Node checks for paths. I doubt compiling `module.exports = require` takes much time in comparison with loading things from the filesystem.

Since this is just a polyfill for `Module.createRequireFromPath`, I'm not too concerned about using `Module._nodeModulePaths` because it could only be removed in Node versions that support `Module.createRequireFromPath` anyway.

Reply:
Ok, sounds good!